app/javascript/components/hello_name.jsx

```diff
@@ -0,0 +1,14 @@
+/* eslint-disable react/prop-types */
+// eslint-disable-next-line no-use-before-define
```

Comment:
What is the purpose of these eslint overrides?

Reply:
I was getting errors from eslint, I've now changed the `.eslintrc.json` to not give these errors and have removed these

Reply:
Before editing the eslint config to stop throwing errors, it's important to know why these errors are being thrown so we know if they're useful or not - after all, eslint is there to help us improve our code.
Could you give a summation of the rules above and justify why it's appropriate to remove them?

Reply:
The main problem I was having was with the `import React from "react";` , this was giving an error saying that I couldn't use "React" before it was defined. 
I understand that a variable should be defined before use but I couldn't understand why it was giving this error for the import, and when I looked for a solution online I found that it may be a bug that eslint may have a different version of react set than my project does, but that didn't fix anything.

As for the prop-types, I'm not 100% sure on this, but  is it asking for a sort of safety net around the props, as in, making sure that they are given and that the component isn't called empty?

Reply:
Okay I found a better way to fix the prop-types problem, and have removed the edit I made to the config